mathics/core/expression.py

```diff
@@ -1150,25 +1193,30 @@ def eval_range(indices):
                 eval_range(range(len(elements)))
                 # rest_range(range(0, 0))
 
+        # Step 2: Build a new expression. We take care not
+        # to evaluate elements, run to_python() on them in
+        # Expression construction, or convert Expresions elements from a tuple to a list
+        # and back if that can be avoided.
+
         if self._elements_fully_evaluated:
-            elements = self._elements
+            new = Expression(
+                head,
+                *self._elements,
+                element_properties={
+                    "_is_flat": self._is_flat,
+                    "_is_ordered": self._is_ordered,
+                    "_elements_fully_evaluated": self._elements_fully_evaluated,
+                }
+            )
+            elements = self.elements
```

Comment:
Even it is the same, shouldn´t be clearer to write `elements = new.elements`? or there is another reason?

Reply:
Since this seems to mean more to you than me (I don't necessarily find one clearer than the other) ab5c21cc has this change.